cmd/kubepkg/templates/latest/deb/kubelet/lib/systemd/system/kubelet-shutdown.service

```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=Shutdown Kubernetes containers before shutting down the system
+Documentation=https://kubernetes.io/docs/home/
+Wants=kubelet.service
+After=kubelet.service
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStop=/usr/bin/crictl rm -af
```

Comment:
OK good question:
- If we use `kubectl delete pod` with node selector, we are depending on a functioning kubelet, so may again generate the deadlock
- We couldn't use Docker CLI here because both Docker/Containerd/CRI-O maybe used

Therefore using `crictl` from my point of view should be the most secure way to communicate with the running CRI engine.

Moreover, shall we inject `cri-tools` package as dependency for kubelet? Or with `ExecStop=-`?

Reply:
Anyway, do we have another way to tell kubelet to kill all its running containers? I had double check the kubelet CLI but couldn't found it.

Reply:
I checked with pkgs.org via openSUSE Tumbleweed:
- [patterns-containers-kubeadm](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/patterns-containers-kubeadm-5.0-22.1.x86_64.rpm.html)
  - [kubernetes-kubeadm](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/kubernetes-kubeadm-1.18.8-6.1.x86_64.rpm.html)
    - [kubernetes1.18-kubeadm](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/kubernetes1.18-kubeadm-1.18.8-2.1.x86_64.rpm.html)
      - [kubernetes1.18-kubelet](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/kubernetes1.18-kubelet-1.18.8-2.1.x86_64.rpm.html)
      - [cri-o](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/cri-o-1.18.3-3.1.x86_64.rpm.html)
      - [cri-tools](https://opensuse.pkgs.org/tumbleweed/opensuse-oss-x86_64/cri-tools-1.18.0-1.4.x86_64.rpm.html)

From this point of view, whenever we install `kubeadm` with distribution package (for k8s master/node setup), we should have both `kubelet`/`cri-o`/`cri-tools` once together.